### PR TITLE
feat: 实现GRU解码器

### PR DESCRIPTION
**Pull request by zhengyangWang1**

### CODE REVIEW

**Comment by Sparidae** [comment]
GRU有问题，如果是每个序列都循环处理其中的单词的话，应该是用nn.GRUCell，他输出每一步的隐层状态，nn.GRU是针对一整个输入序列计算的，返回的是序列的每一个时间步的历史隐层状态列表和最后一个时间步的输出，详细请看文档
[torch.nn.GRU](https://pytorch.org/docs/stable/generated/torch.nn.GRU.html#torch.nn.GRU)
[torch.nn.GRUCell](https://pytorch.org/docs/stable/generated/torch.nn.GRUCell.html#torch.nn.GRUCell)

**Review by Sparidae — CHANGES_REQUESTED** [review]
GRU有问题，如果是每个序列都循环处理其中的单词的话，应该是用nn.GRUCell，他输出每一步的隐层状态，nn.GRU是针对一整个输入序列计算的，返回的是序列的每一个时间步的历史隐层状态列表和最后一个时间步的输出，详细请看文档
[torch.nn.GRU](https://pytorch.org/docs/stable/generated/torch.nn.GRU.html#torch.nn.GRU)
[torch.nn.GRUCell](https://pytorch.org/docs/stable/generated/torch.nn.GRUCell.html#torch.nn.GRUCell)

**Comment by zhengyangWang1** [comment]
GRU接受（L, N, H）作为输入，x = torch.cat((img, cap), dim=-1).unsqueeze(0)这里加了unsqueeze(0), 把第0维的维度设为1，对应GRU输入的L，也就是只有一个时间步

**Review by Sparidae — APPROVED** [review]
